src/main/java/de/tum/in/www1/artemis/security/ArtemisSecurityUser.java

```diff
@@ -0,0 +1,221 @@
+package de.tum.in.www1.artemis.security;
+
+import java.io.Serializable;
+import java.util.*;
+
+import org.springframework.security.core.CredentialsContainer;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.SpringSecurityCoreVersion;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.util.Assert;
+import org.springframework.util.ObjectUtils;
+
+public class ArtemisSecurityUser implements UserDetails, CredentialsContainer {
```

Comment:
we should avoid copying Spring classes, this will lead to high maintenance efforts in case Spring changes something

Reply:
Interesting thing: I'm not quite sure what I previously changed that let everything fail (which was why I introduced this class), but now it works even without this class.
The reason is that for internal users, it works as they don't have a null password. This will throw an exception for external users, but this exception will only cause the external provider to kick in, which is exactly what we want.

Therefore, I removed this class again.